### PR TITLE
NIFI-5909 PutElasticsearchHttpRecord doesn't allow to customize the timestamp format

### PR DESCRIPTION
**Pull request by SavtechSolutions**
added optional settings for date, time, and timestamp formats used to write Records to Elasticsearch

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by SavtechSolutions** [comment]
No unit tests have been added/modified, as existing tests verify only the URL used to call ES, but not the content. I'm thinking of adding a callback hook to the realRequest inside PutElasticsearchHttpRecordTestProcessor.createElasticsearchClient() to do any custom content assertions

**Comment by SavtechSolutions** [comment]
Yes, I do plan to add them - time permitting, as there's some infrastructure coding I need to do first, to make content assertions possible.

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchHttpRecord.java [hunk 1]
```diff
@@ -178,6 +178,36 @@
             .required(true)
             .build();
 
+    static final PropertyDescriptor DATE_FORMAT = new PropertyDescriptor.Builder()
+            .name("put-es-record-date-format")
+            .displayName("Date Format")
+            .description("Custom date format to use when converting fields of date type " +
```
[the remark this answers is not in the document]
Comment (SavtechSolutions) [reply nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchHttpRecord.java]: Just to confirm, by "reusing" you mean copying what's in DateTimeUtils into my PDs, not to reference existing PDs? They do share a lot of common, but have different fallback values (millisecond longs in CSV vs RecordFieldType.*.getDefaultFormat() for ES), so I don't think I can simply reference the existing PDs

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/pom.xml [hunk 2]
```diff
@@ -122,6 +122,11 @@ language governing permissions and limitations under the License. -->
             <artifactId>jackson-databind</artifactId>
             <version>${jackson.version}</version>
         </dependency>
+	    <dependency>
```
Comment (bdesert) [inline nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/pom.xml]: @SavtechSolutions ,
could you please change tabs to spaces here as well?

**Pull request closed without merging**